sklearn/decomposition/pca.py

```diff
@@ -177,35 +221,60 @@ class PCA(BaseEstimator, TransformerMixin):
     via the score and score_samples methods.
     See http://www.miketipping.com/papers/met-mppca.pdf
 
-    Due to implementation subtleties of the Singular Value Decomposition (SVD),
-    which is used in this implementation, running fit twice on the same matrix
-    can lead to principal components with signs flipped (change in direction).
-    For this reason, it is important to always use the same estimator object to
-    transform data in a consistent fashion.
+    For svd_solver == 'arpack', refer to `scipy.sparse.linalg.svds`.
+
+    For svd_solver == 'randomized', see:
+    `Finding structure with randomness: Stochastic algorithms
+    for constructing approximate matrix decompositions Halko, et al., 2009
+    (arXiv:909)`
+    `A randomized algorithm for the decomposition of matrices
+    Per-Gunnar Martinsson, Vladimir Rokhlin and Mark Tygert`
+
 
     Examples
     --------
-
     >>> import numpy as np
     >>> from sklearn.decomposition import PCA
     >>> X = np.array([[-1, -1], [-2, -1], [-3, -2], [1, 1], [2, 1], [3, 2]])
     >>> pca = PCA(n_components=2)
     >>> pca.fit(X)
-    PCA(copy=True, n_components=2, whiten=False)
+    PCA(copy=True, iterated_power=4, n_components=2, random_state=None,
+      svd_solver='auto', tol=0.0, whiten=False)
+    >>> print(pca.explained_variance_ratio_) # doctest: +ELLIPSIS
+    [ 0.99244...  0.00755...]
+
+    >>> pca = PCA(n_components=2, svd_solver='full')
```

Comment:
I guess my idea at the time was to show that we can obtain the same `explained_variance_ratio_` running PCA with different solvers. Let me know if you would rather get rid of those.